application/schemas/templates/components.py

```diff
@@ -28,8 +37,41 @@ class Component(BaseModel):
                     'a higher priority.',
     )
 
+
+class HttpRequest(BaseModel):
+    """
+    Deploy schema for HTTP type components.
+    """
+    url: HttpUrl = Field(description='URL for HTTP deployment.', example='http://example.com/path/to/deploy')
+    method: ComponentTypeHttpMethods = Field(
+        default=ComponentTypeHttpMethods.get,
+        description='HTTP method for deployment',
+        example=ComponentTypeHttpMethods.post)
+    headers: dict | list[dict] | None = Field(
+        description='Headers for HTTP deployment', example={
+            'Content-Type': 'application/json'})
+    parameters: dict | list[dict] | None = Field(
+        description='Parameters for HTTP deployment', example=[{'param1': 'value1'}])
+
+    class Config:
+        extra = Extra.forbid
+
+
+class ComponentHttp(BaseModel):
+    """
+    Application component HTTP schema.
+    """
+    name: constr(min_length=1, strip_whitespace=True) = Field(description='Component Name', example='jenkins')
+    type: Literal[ComponentTypes.http]
+    enabled: bool | TemplateVariable | None = Field(description='Is component must be installed.', default=True)
+    create: HttpRequest = Field(description='Creation details for HTTP type components')
+    delete: HttpRequest = Field(description='Deletion details for HTTP type components')
+    health: HttpRequest | None = Field(description='Health check for HTTP type components')
+
     class Config:
         extra = Extra.forbid
 
-    def __hash__(self) -> str:
-        return hash(self.name)
+
+Component = Annotated[ComponentHelm |
```

Comment:
I think `Component` better to rename to `Components` because it represent multiple component not one.